### ISSUE TITLE
Question: What can you do with Trace Result?

### ISSUE THREAD

**Issue by simongis** [Esri/solutions-widget-utility-network-trace#21]
Apologies if this belongs elsewhere such as GeoNet.

Working with the Naperville Electric data, I can perform some upstream/downstream traces just fine.

If I wanted to showcase this widget to its full potential, can you perhaps provide me some information on what you can do to really showcase the capability of this widget?

- I see you can export the trace result to a CSV directly from a widget
- I think the trace result is a temporary graphics layer?  Can it be exported as a new item?
- I was hoping it would appear in other widgets - e.g. Chart & Infographic 
- Or I was hoping it might select features in the existing layers, but the attribute table does not recognise that features have been selected.

Just looking for some tips.  Perhaps there is a video I might have missed that highlights more about its capabilities?

**Comment by MikeMillerGIS** [comment]
The widget currently is just showcasing that you can call the trace framework of the UN.  The results and the UX with them is very limited at the moment.  You provided some good suggestions, lets log them as issues and we can work on adding them in the future.  Thank you for your feedback

**Comment by MikeMillerGIS** [comment]
closing as I have created individual issues for your request

**Comment by aventham** [comment]
Is this widget designed to display the trace results as graphics? I'm getting the correct results, but only as CSV data via the "export" link.

**Comment by MikeMillerGIS** [comment]
Yes, displays as graphics.  You can configure each result to save to a layer

**Comment by aventham** [comment]
Thanks for confirming!
Unfortunately I'm not getting any result graphics. Perhaps this is a knock-on effect from the "The layer, Water Utility Network, cannot be added to the map" error I always get when the application loads (although there's seemingly no problem with loading this layer in the web map author).
The trace operation runs impressively quickly, but all I get is the option to export a CSV of the results.
I expect it's just a problem with my config rather than a code bug - so a few lines of guidance on the necessary steps to configure the web map and widget would be a great help and probably clear this up.
Unless 27700 co-ordinate system / projection isn't supported yet?

**Comment by MikeMillerGIS** [comment]
Some screen shots would help.  Also, is anything reported in the console?

**Comment by aventham** [comment]
Sure - here are a couple of browser screenshots (not really that much to see though, without any graphics!), and how the same trace results look in ArcGIS Pro. Nothing interesting in the browser console as far as I can see.

![trace01](https://user-images.githubusercontent.com/38691836/70901259-5b263a00-1ff2-11ea-819e-3f0d01cfadb5.png)
![trace02](https://user-images.githubusercontent.com/38691836/70901261-5b263a00-1ff2-11ea-8b0f-c47131c19fd4.png)
![trace03](https://user-images.githubusercontent.com/38691836/70901262-5b263a00-1ff2-11ea-8468-dbc4b93ddb60.png)

**Comment by MikeMillerGIS** [comment]
Ah my fault, I for some reason was thinking of our GN network Trace widget.  This widget only provides csv download.  The results should be drawn though.  

If you use fiddler, can you ensure the Trace is completing?  The widget needs to get the features from the map as the Trace does not return features, just IDs.

**Comment by MikeMillerGIS** [comment]
Oh, change you domain from DMA to system too

**Comment by aventham** [comment]
Are you saying it will only return graphics when tracing on the system tier?
So far we've only set up subnetwork controllers on the DMA tier.
But I can see in Chrome dev tools that the trace request gets a complete and valid JSON response back.

**Comment by MikeMillerGIS** [comment]
If the dma tier is working, then no issue.  The widget is not finding the features in the map for some reason

**Comment by aventham** [comment]
I imagine it's probably something to do with our OSGB map/data spatial reference then?
I've tried changing the wkid and latestwkid to 27700 in config.json but I'm still seeing 102100 and 3857 in the outSR portion of the post-trace query responses.

**Comment by aventham** [comment]
Aha! I've "fixed" it by altering line 132 of utilitynetwork.js to force the outSR to 27700.
